---
title: "Deseq2_eachtime"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

```{r}
library(DESeq2)
library("RColorBrewer")
library('pheatmap')
library('dplyr')
library(DEGreport)
library("ggplot2")
library('tidyverse')
library(biomaRt)
ensembl = useMart("ensembl", dataset="hsapiens_gene_ensembl") #uses human ensembl annotations
library("AnnotationDbi")
library("org.Hs.eg.db")
library("BiocParallel")
register(MulticoreParam(4))
library(EnhancedVolcano)
```

```{r}

counts<-read.table(file='countmatrix_aug_col4.csv', header=TRUE,row.names=1)
```

```{r}
## Performing pairwise comparisions to obtain differentially expressed genes between the control and treatment group for hour2, 4, 6 and 8 hours. Below is an example code for hour 4 comparison. All comparisons can be made similarly with appropriate sample names.

samples <- read.table(file='path_to_your_directory_where_the_design_of_the_expt_exists/sample4h.txt', header=TRUE,sep='\t',row.names=1,colClasses=c('factor','factor','factor')) 

columorder = c("M14C", "M24C",  "M34C", "M44C", "M54C", "M64C",    
               "M14F", "M24F",  "M34F", "M44F", "M54F", "M64F")
ordcounts <- counts[columorder]
dds <- DESeqDataSetFromMatrix(countData = ordcounts,
                              colData = samples,
                              design = ~1+donor+group)
dds <- estimateSizeFactors(dds)
nc <- counts(dds, normalized=TRUE)
filter <- rowSums(nc >= 10) >= 6 #Gene count is greater than 10 in at least 6 samples
dds <- dds[filter,]
dds<-DESeq(dds)

sep4h = results(dds, name="group_F4_vs_C4",alpha=0.05)
summary(sep4h)
ensgene=getBM(attributes=c('ensembl_gene_id_version', 'ensembl_gene_id', 'hgnc_symbol', 'wikigene_description', 'chromosome_name', 'start_position', 'end_position'), 
                       filters = 'ensembl_gene_id', 
                       values = rownames(sep4h), 
                       mart = ensembl)
out4 <- tibble::rownames_to_column(as.data.frame(sep4h), "ensembl_gene_id")
final_4h_FC <- merge(out4,ensgene, by.x = "ensembl_gene_id", by.y = "ensembl_gene_id")
colorder <- c("ensembl_gene_id", "hgnc_symbol", "log2FoldChange", "lfcSE", "pvalue", "padj", "baseMean","stat", 'wikigene_description', "ensembl_gene_id_version", "chromosome_name", "start_position", "end_position")
ordered4h_FC <- final_4h_FC[colorder]
wanted = c("M14C", "M24C",  "M34C", "M44C", "M54C", "M64C", "M14F", "M24F", "M34F", "M44F", "M54F", "M64F")
wanted_cts = as.data.frame(counts(dds)[,wanted])
cts4_id <- tibble::rownames_to_column(wanted_cts, "ensembl_gene_id")
export4h_FC <- merge(ordered4h_FC, cts4_id, by.x = "ensembl_gene_id", by.y = "ensembl_gene_id" )
sorted_4H_FC <- export4h_FC[with(ordered4h_FC, order(padj)),]
write.csv(sorted_4H_FC,file="/Users/banditaadhikari/Desktop/backup_to_googleDrive/Aspergillus/final_de/michael_advice/col4_aug/4h_FC_separate_4c.csv")
```

```{r}
library('EnhancedVolcano')
jpeg('/Users/banditaadhikari/Desktop/backup_to_googleDrive/Aspergillus/final_de/michael_advice/col4_aug/4h_FC_separate_c4.jpg')
EnhancedVolcano(sorted_4H_FC,
    lab = sorted_4H_FC$hgnc_symbol,
    x = 'log2FoldChange',
    y = 'padj', # plotted adjusted p value
     selectLab = c(''), # you can label any gene name here, but i dont want any, so kept it blank
    ylim = c(-0.5, 8),
    xlim = c(-4, 4),
    title = 'Macrophage vs Macrophage+Fungus, 2 hours',
    FCcutoff = 1,
    transcriptPointSize = 1.5,
    transcriptLabSize = 3.0,
    col=c('black', 'black', 'black', 'red3'),
    colAlpha = 1) #lab = rownames(sep2h),pCutoff = 10e-5,
dev.off()
```

